pex/bin/pex.py

```diff
@@ -477,6 +487,13 @@ def build_pex(args, options, resolver_option_builder):
   for requirements_txt in options.requirement_files:
     resolvables.extend(requirements_from_file(requirements_txt, resolver_option_builder))
 
+  for constraints_txt in options.constraint_files:
+    constraints = []
+    for r in requirements_from_file(constraints_txt, resolver_option_builder):
```

Comment:
pip states the constraints format is identical to requirements:

https://pip.pypa.io/en/stable/user_guide/#constraints-files

so I figured the way to guarantee that in pex was to share the requirements code.